connect-button/src/main/java/com/ifttt/connect/ui/AnalyticsEventUploader.java

```diff
@@ -0,0 +1,55 @@
+package com.ifttt.connect.ui;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+import java.io.IOException;
+import java.util.List;
+import retrofit2.Response;
+
+public class AnalyticsEventUploader extends Worker {
+
+    private static String anonymousId;
+    private AnalyticsManager analyticsManager;
+    private static final int MAX_RETRY_COUNT = 3;
+
+    public AnalyticsEventUploader(@NonNull Context context, @NonNull WorkerParameters params) {
+        super(context, params);
+        analyticsManager = AnalyticsManager.getInstance(context.getApplicationContext());
+        anonymousId = AnalyticsPreferences.getInstance(context).getAnonymousId();
+    }
+
+    @Override
+    @NonNull
+    public Result doWork() {
+        Result result;
+        try {
+            int queueSize = analyticsManager.getQueueSize();
+            List<AnalyticsEventPayload> list = analyticsManager.performRead(queueSize);
```

Comment:
Sure, the read method can be re-written to work that way. 

But the intention of doing this is to check the queue size at the beginning, then read and remove only those many elements. There is an api call between read and remove, during which the enqueue operation could have acquired the lock and changed the queue content.  We want to remove only those number of elements that were read earlier. So doing it this way is clearer to read imo.

Reply:
We will know the size when we read the queue for the payloads, right? So here we will be able to get the size of the items by doing `list.size()`, instead of having to call `getQueueSize`.

Reply:
oops, yes.